internal/provider/kubernetes/status.go

```diff
@@ -66,16 +66,20 @@ func (r *gatewayAPIReconciler) subscribeAndUpdateStatus(ctx context.Context, ext
 				r.statusUpdater.Send(Update{
 					NamespacedName: key,
 					Resource:       new(gwapiv1.HTTPRoute),
-					Mutator: MutatorFunc(func(obj client.Object) client.Object {
+					Mutator: MutatorFunc(func(obj client.Object) bool {
```

Comment:
the API is for mutating obects but is now returning a `bool` ? this doesnt seem right

Reply:
changed this api to boolean

Reply:
What is the new definition for the mutator function

Reply:
https://github.com/envoyproxy/gateway/blob/5b9553243da45d95c398e4ffde3551efd5ce1f07/internal/provider/kubernetes/status_updater.go#L38-L53